tilequeue/postgresql.py

```diff
@@ -1,69 +1,44 @@
 from itertools import cycle
-from psycopg2.pool import ThreadedConnectionPool
 from psycopg2.extras import register_hstore, register_json
+import psycopg2
 import threading
-import ujson as json
-
-
-class DatabaseCycleConnectionPool(object):
-
-    """
-    Maintains a psycopg2 ThreadedConnectionPool for each of the
-    given dbnames. When a client requests a set of connections,
-    all of those connections will come from the same database.
-    """
-
-    def __init__(self, min_conns_per_db, max_conns_per_db, dbnames, conn_info):
-        self._pools = []
-        self._conns_to_pool = {}
-
-        for dbname in dbnames:
-            pool = ThreadedConnectionPool(
-                min_conns_per_db,
-                max_conns_per_db,
-                dbname=dbname,
-                **conn_info
-            )
-            self._pools.append(pool)
-
-        self._pool_cycle = cycle(self._pools)
-        self._lock = threading.Lock()
-
-    def get_conns(self, n_conns):
-        conns = []
-
-        try:
-            with self._lock:
-                pool_to_use = next(self._pool_cycle)
-                for _ in range(n_conns):
-                    conn = pool_to_use.getconn()
-
-                    conn.set_session(readonly=True, autocommit=True)
-                    register_json(conn, loads=json.loads, globally=True)
-                    register_hstore(conn, globally=True)
-
-                    self._conns_to_pool[id(conn)] = pool_to_use
-                    conns.append(conn)
-                assert len(conns) == n_conns, \
-                    "Couldn't collect enough connections"
-        except:
-            if conns:
-                self.put_conns(conns)
-                conns = []
-            raise
-
+import ujson
+
+
+class DBAffinityConnectionsNoLimit(object):
```

Comment:
We didn't use the other database pools, so I removed them.

The ujson stuff is there. The globally=True thing I wanted to leave off in case it caused the connection problem.